--- a/units/console-getty.service.m4.in
+++ b/units/console-getty.service.m4.in
@@ -16,7 +16,7 @@ After=rc-local.service
 Before=getty.target
 
 [Service]
-ExecStart=-/sbin/agetty --noclear --keep-baud console 115200,38400,9600 $TERM
+ExecStart=-/usr/sbin/agetty --noclear --keep-baud console 115200,57600,38400,9600 $TERM
 Type=idle
 Restart=always
 RestartSec=0
--- a/units/container-getty@.service.m4.in
+++ b/units/container-getty@.service.m4.in
@@ -17,7 +17,7 @@ IgnoreOnIsolate=yes
 ConditionPathExists=/dev/pts/%I
 
 [Service]
-ExecStart=-/sbin/agetty --noclear --keep-baud pts/%I 115200,38400,9600 $TERM
+ExecStart=-/usr/sbin/agetty --noclear --keep-baud pts/%I 115200,57600,38400,9600 $TERM
 Type=idle
 Restart=always
 RestartSec=0
--- a/units/getty@.service.m4
+++ b/units/getty@.service.m4
@@ -27,7 +27,7 @@ ConditionPathExists=/dev/tty0
 
 [Service]
 # the VT is cleared by TTYVTDisallocate
-ExecStart=-/sbin/agetty --noclear %I $TERM
+ExecStart=-/usr/sbin/agetty --noclear %I $TERM
 Type=idle
 Restart=always
 RestartSec=0
--- a/units/serial-getty@.service.m4
+++ b/units/serial-getty@.service.m4
@@ -22,7 +22,7 @@ Before=getty.target
 IgnoreOnIsolate=yes
 
 [Service]
-ExecStart=-/sbin/agetty --keep-baud 115200,38400,9600 %I $TERM
+ExecStart=-/usr/sbin/agetty --login-program /bin/ash --login-options "--login" --skip-login --keep-baud 115200,57600,38400,9600 %I $TERM
 Type=idle
 Restart=always
 UtmpIdentifier=%I
