--- a/include/sane/sanei.h
+++ b/include/sane/sanei.h
@@ -155,6 +155,16 @@ extern SANE_Status sanei_check_value (co
 
 extern SANE_Status sanei_constrain_value (const SANE_Option_Descriptor * opt,
 					  void * value, SANE_Word * info);
+#if !defined( __i386__) && !defined(__arm__) || defined(__MUSL_GLIBC_TYPES_H)
+static unsigned char inb(unsigned int port)
+{
+       return 0;
+}
+
+static void outb(unsigned char value, unsigned int port)
+{
+}
+#endif
 
 
 #endif /* sanei_h */
--- a/sanei/sanei_udp.c
+++ b/sanei/sanei_udp.c
@@ -46,6 +46,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <fcntl.h>
+#include <sys/types.h>
 
 #ifdef HAVE_WINSOCK2_H
 #include <winsock2.h>
